addons/website_forum/models/forum_post.py

```diff
@@ -277,26 +284,16 @@ def _search_can_view(self, operator, value):
         user = self.env.user
         # Won't impact sitemap, search() in converter is forced as public user
         if self.env.is_admin():
-            return [(1, '=', 1)]
-
-        req = """
-            SELECT p.id
-            FROM forum_post p
-                   LEFT JOIN res_users u ON p.create_uid = u.id
-                   LEFT JOIN forum_forum f ON p.forum_id = f.id
-            WHERE
-                (p.create_uid = %s and f.karma_close_own <= %s)
-                or (p.create_uid != %s and f.karma_close_all <= %s)
-                or (
-                    u.karma > 0
-                    and (p.active or p.create_uid = %s)
-                )
-        """
+            return [(1, operator, 1)]
 
-        op = 'inselect' if operator == '=' else "not inselect"
-
-        # don't use param named because orm will add other param (test_active, ...)
-        return [('id', op, (req, (user.id, user.karma, user.id, user.karma, user.id)))]
+        post_domain = expression.OR([
+            [('create_uid', '=', user.id)],
+            [('state', 'in', ['offensive', 'pending']), ('forum_id.karma_moderate', '<=', user.karma)],
+            [('state', '=', 'close'), ('forum_id.karma_close_all', '<=', user.karma)],
+            [('state', 'in', ('active', 'flagged')), ('create_uid.karma', '>', 0)],
+        ])
+        posts_ids = self.sudo().search(post_domain).ids
```

Comment:
I guess you need to search to bypass access rules ? (worth a comment as it's not obvious)
also to not trigger a SQL query, you can do

```suggestion
        posts_ids = self.sudo()._search(post_domain)

Reply:
Check for `_search`. I must confess I don't know why the domain alone didn't work, access rules didn't seem in the way?